client/src/components/Dashboard/index.js

```diff
@@ -15,13 +18,149 @@ import YourFavorites from "./YourFavorites";
+import firebase from "firebase";
+
+const renderInputComponent = inputProps => {
+  const { classes, inputRef = () => { }, ref, ...other } = inputProps;
+
+  return (
+    <TextField
+      fullWidth
+      InputProps={{
+        inputRef: node => {
+          ref(node);
+          inputRef(node);
+        },
+        classes: {
+          input: classes.input
+        }
+      }}
+      {...other}
+    />
+  );
+};
+
+const renderSuggestion = (suggestion, { query, isHighlighted }) => {
+  const matches = match(suggestion.label, query);
+  const parts = parse(suggestion.label, matches);
+
+  return (
+    <MenuItem selected={isHighlighted} component="div">
+      <div>
+        {parts.map((part, index) =>
+          part.highlight ? (
+            <span key={String(index)} style={{ fontWeight: 500 }}>
+              {part.text}
+            </span>
+          ) : (
+              <strong key={String(index)} style={{ fontWeight: 300 }}>
+                {part.text}
+              </strong>
+            )
+        )}
+      </div>
+    </MenuItem>
+  );
+};
+
+const getSuggestions = value => {
+  const inputValue = deburr(value.trim()).toLowerCase();
+  const inputLength = inputValue.length;
+  let count = 0;
+
+  return inputLength === 0
+    ? []
+    : suggestions.filter(suggestion => {
+      const keep =
+        count < 5 &&
+        suggestion.label.slice(0, inputLength).toLowerCase() === inputValue;
+
+      if (keep) {
+        count += 1;
+      }
+
+      return keep;
+    });
+};
+
+const getSuggestionValue = suggestion => {
+  return suggestion.label;
+};
+
 class Dashboard extends Component {
   state = {
+    value: 0,
+    data: [],
+    ticker: "",
+    single: "",
+    popper: "",
+    suggestions: [],
+    uid: firebase.auth().currentUser.uid,
     checked: false
   };
+
+  handleSuggestionsFetchRequested = ({ value }) => {
+    this.setState({
+      suggestions: getSuggestions(value)
+    });
+  };
+
+  handleSuggestionsClearRequested = () => {
+    this.setState({
+      suggestions: []
+    });
+  };
+
+  handleChange = name => (event, { newValue }) => {
+    this.setState({
+      [name]: newValue
+    });
+  };
+  onKeyDown = e => {
+    if (e.key === "Enter") {
+      this.handleChange();
+    }
+  };
+
+  onSuggestionSelected = (
+    event,
+    { suggestion, suggestionValue, suggestionIndex, sectionIndex, method }
+  ) => {
+    const newSymbol = {
+      symbol: suggestionValue,
+      uid: this.state.uid
+    };
+    axios.post('https://pickemm.herokuapp.com/api/favorites', newSymbol)
+      .then(response => {
+        this.setState({
+          newSymbol: { symbol: '', uid: '' }
+        });
+        window.location.reload();
+      })
+      .catch(err => { console.log("we've encountered an error") })
+  };
+
   render() {
     const { checked } = this.state;
     const { classes } = this.props;
+
+    const autosuggestProps = {
+      renderInputComponent,
+      onKeyPress: this.onKeyDown,
```

Comment:
This is awesome!!! Thanks Willie after reviewing this I like that I can now press enter